src/modules/dashboard/components/RecoveryModeDialog/RecoveryModeDialogForm.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { ColonyRole } from '@colony/colony-js';
+
+import Button from '~core/Button';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Annotations } from '~core/Fields';
+import Heading from '~core/Heading';
+import PermissionsLabel from '~core/PermissionsLabel';
+import PermissionRequiredInfo from '~core/PermissionRequiredInfo';
+
+import { useLoggedInUser, Colony } from '~data/index';
+import { useTransformer } from '~utils/hooks';
+
+import { getAllUserRoles } from '../../../transformers';
+import { canEnterRecoveryMode } from '../../../users/checks';
+
+import styles from './RecoveryModeDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'admin.RecoveryModeDialog.RecoveryModeDialogForm.title',
+    defaultMessage: 'Enter Recovery mode',
+  },
+  description1: {
```

Comment:
I would change those names to something more descriptive, for example:
description1 -> recoveryModeDescription
description2 -> leavingRecoveryModeDescription